app/src/main/java/com/example/weather/CityPickerFragment.kt

```diff
@@ -35,9 +37,34 @@ class CityPickerFragment : Fragment() {
         cityPickerAdapter = CityPickerAdapter(context)
         recyclerView.adapter = cityPickerAdapter
         viewModel.cities.observe(viewLifecycleOwner) {
-            cityPickerAdapter.updateCityData(it)
+            when(it) {
+                is StateOverview.Error -> displayOverviewError()
+                is StateOverview.Loading -> displayOverviewLoading()
+                is StateOverview.Success -> displayOverviewSuccess(it.data)
+            }
         }
         viewModel.getCityInfo()
     }
 
+    private fun displayOverviewError() {
+        binding.recyclerView.visibility = View.GONE
+        binding.loading.visibility = View.GONE
+        binding.errorMessage.text = getString(R.string.error_message)
+        binding.errorMessage.visibility = View.VISIBLE
+        binding.snag.visibility = View.VISIBLE
+    }
+
+    private fun displayOverviewLoading() {
+        binding.recyclerView.visibility = View.GONE
```

Comment:
I would add a line to set the error stuff to GONE so in the scenario you go from an error screen to success screen, both the error stuff and data aren't shown at the same time